composables/cms/useCmsElementPreload.ts

```diff
@@ -0,0 +1,44 @@
+import type { Schemas } from '@shopware/api-client/api-types';
+
+export function useCmsElementPreload(cmsPage?: Schemas['CmsPage']) {
+    const { getBlockClasses } = useCmsUtils();
+    const runtimeConfig = useRuntimeConfig();
+
+    const getPreloadElementId = (cmsPage: Schemas['CmsPage']): string | undefined => {
+        const isPreloadingEnabled = runtimeConfig.public.pond.preloading.enabled;
+        const preloadingClass = runtimeConfig.public.pond.preloading.imagePreloadingClass;
+
+        if (!cmsPage || !isPreloadingEnabled) {
+            return;
+        }
+
+        let cmsElementToPreload: Schemas['CmsSlot'] | undefined;
+
+        cmsPage.sections.forEach((section: Schemas['CmsSection']) => {
+            section.blocks.forEach((block: Schemas['CmsBlock']) => {
+                const blockClasses = getBlockClasses(block);
+                const hasPreloadingClass = blockClasses.includes(preloadingClass);
+
+                if (hasPreloadingClass && !cmsElementToPreload) {
+                    cmsElementToPreload = block.slots[0];
+                }
+            });
+        });
+
+        if (cmsElementToPreload) {
+            return cmsElementToPreload.id;
+        }
+
+        return cmsPage.sections[0]?.blocks[0]?.slots[0]?.id;
```

Comment:
Why do you iterate over all the sections and blocks here? Can't you just access the blockClasses the same way you return the id at the end, since only the first section / block is relevant? Something like this:

```suggestion
        const blockClasses = getBlockClasses(cmsPage.sections[0]?.blocks[0]);
        // probably add a null check or smth here
        const hasPreloadingClass = blockClasses.includes(preloadingClass);

        if (hasPreloadingClass && !cmsElementToPreload) {
            cmsElementToPreload = cmsPage.sections[0]?.blocks[0]?.slots[0];
        }

        if (cmsElementToPreload) {
            return cmsElementToPreload.id;
        }

        return cmsPage.sections[0]?.blocks[0]?.slots[0]?.id;
```

Reply:
we need to go through all the blocks because the functionality developed is like this:
1. You can configure a block to be preloaded if you add a custom css class ( that is configurable )
2. If no class is found then the first element from the first block is preloaded

Reply:
Alright, did you test for potential slow downs on big content pages with lots of sections / blocks? We don't want the performance gain by preloading to be negated by long execution times

Reply:
Ideally we would have rough estimates for these 3 values:
- Large content page with no adjustments
- Large content page with preloading and iteration of all blocks / sections
- Large content page with preloading of first element only (perhaps up to first three elements?) and no iteration

Reply:
Yea, I expected as much but still wanted to make sure. Thanks for verifying! Looks good to me 👍